src/Amalgam/AssetManager.h

```diff
@@ -107,6 +107,18 @@ class AssetManager
 	inline bool IsEntityDirectlyPersistent(Entity *entity)
 	{	return persistentEntities.find(entity) != end(persistentEntities);	}
 
+	//sets the entity's persistent path
+	inline void SetEntityPersistentPath(Entity *entity, std::string &resource_path)
+	{
+	#ifdef MULTITHREAD_INTERFACE
+		Concurrency::WriteLock lock(persistentEntitiesMutex);
+	#endif
+		if(resource_path == "")
```

Comment:
I see you switched to use `.empty()` in a few other spots with this PR. Might wanna do that here as well if that's the way we are leaning. No big deal as far as I know though